src/components/geolocation/GeolocationDirective.js

```diff
@@ -47,6 +57,23 @@
           }
         });
 
+        // Scope watchers
+        scope.$watch('tracking', function(tracking) {
```

Comment:
This watches whether the tracking is triggererd (happens when the swicht on/off the geolocation/orientation. The `change:heading' events is called at every change of event (hence very often), in my opinion we should have as less as possible code inside it, and keep like that...